pkg/pbq/pbq.go

```diff
@@ -0,0 +1,144 @@
+package pbq
+
+import (
+	"context"
+	"errors"
+	"github.com/numaproj/numaflow/pkg/isb"
+	"github.com/numaproj/numaflow/pkg/pbq/store"
+	"github.com/numaproj/numaflow/pkg/shared/logging"
+	"go.uber.org/zap"
+	"time"
+)
+
+var EOF error = errors.New("error while reading, EOF")
+var COBErr error = errors.New("error while writing to pbq, pbq is closed")
+
+type PBQ struct {
+	Store       store.Store
+	output      chan *isb.Message
+	cob         bool // cob to avoid panic in case writes happen after close of book
+	partitionID string
+	options     *store.Options
+	isReplaying bool
+	manager     *Manager
+	log         *zap.SugaredLogger
+}
+
+// NewPBQ accepts size and store and returns new PBQ
+func NewPBQ(ctx context.Context, partitionID string, persistentStore store.Store, pbqManager *Manager, options *store.Options) (*PBQ, error) {
+
+	// output channel is buffered to support bulk reads
+	p := &PBQ{
+		Store:       persistentStore,
+		output:      make(chan *isb.Message, options.BufferSize()),
+		cob:         false,
+		partitionID: partitionID,
+		options:     options,
+		manager:     pbqManager,
+		log:         logging.FromContext(ctx).With("PBQ", partitionID),
+	}
+
+	p.manager.Register(partitionID, p)
```

Comment:
Instead of calling manager to register, is it better to move `New` function to the manager?